akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -412,6 +456,49 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
       allocationStrategy: ShardAllocationStrategy,
       handOffStopMessage: Any): ActorRef = {
 
+    internalStart(
+      typeName,
+      _ => entityProps,
+      settings,
+      extractEntityId = {
+        case msg if messageExtractor.entityId(msg) ne null =>
+          (messageExtractor.entityId(msg), messageExtractor.entityMessage(msg))
+      },
+      extractShardId = msg => messageExtractor.shardId(msg),
+      allocationStrategy = () => allocationStrategy,
+      handOffStopMessage = handOffStopMessage)
+  }
+
+  /**
+   * Java/Scala API: Register a named entity type by defining the [[akka.actor.Props]] of the entity actor
+   * and functions to extract entity and shard identifier from messages. The [[ShardRegion]] actor
+   * for this type can later be retrieved with the [[#shardRegion]] method.
+   *
+   * This method will start a [[ShardRegion]] in proxy mode when there is no match between the
+   * node roles and the role specified in the [[ClusterShardingSettings]] passed to this method.
+   *
+   * Some settings can be configured as described in the `akka.cluster.sharding` section
+   * of the `reference.conf`.
+   *
+   * @param typeName the name of the entity type
+   * @param entityProps the `Props` of the entity actors that will be created by the `ShardRegion`
+   * @param settings configuration settings, see [[ClusterShardingSettings]]
+   * @param messageExtractor functions to extract the entity id, shard id, and the message to send to the
+   *   entity from the incoming message, see [[ShardRegion.MessageExtractor]]
+   * @param allocationStrategy possibility to use a custom shard allocation and
+   *   rebalancing logic
+   * @param handOffStopMessage the message that will be sent to entities when they are to be stopped
+   *   for a rebalance or graceful shutdown of a `ShardRegion`, e.g. `PoisonPill`.
+   * @return the actor ref of the [[ShardRegion]] that is to be responsible for the shard
+   */
+  def start(
+      typeName: String,
+      entityProps: Props,
+      settings: ClusterShardingSettings,
+      messageExtractor: ShardRegion.MessageExtractor,
+      allocationStrategy: () => ShardAllocationStrategy,
+      handOffStopMessage: Any): ActorRef = {
+
```

Comment:
I find 3 `start with `: ShardAllocationStrategy` param but only 2 with `() => ShardAllocationStrategy`. Missing one overload?

Reply:
WIll look at that if we don't revert the API changes